internal/postrs/pos_verifier.go

```diff
@@ -70,14 +71,24 @@ func VerifyPos(dataDir string, scryptParams ScryptParams, o ...VerifyPosOptionsF
 	defer C.free(unsafe.Pointer(dataDirPtr))
 
 	result := C.verify_pos(dataDirPtr, (*C.uint32_t)(opts.fromFile), (*C.uint32_t)(opts.toFile), C.double(opts.fraction), scryptParams)
-	switch result {
-	case C.Ok:
+	switch result.tag {
+	case C.VerifyResult_Ok:
 		return nil
-	case C.Invalid:
-		return ErrInvalidPos
-	case C.InvalidArgument:
+	case C.VerifyResult_Invalid:
+		result := castBytes[C.VerifyPosResult_Invalid_Body](result.anon0[:])
+		return fmt.Errorf("%w: file %d contains invalid label at offset %d", ErrInvalidPos, result.file, result.offset)
```

Comment:
That seems a) dangerous and b) very platform / compiler specific; i.e. just because it works on windows/amd64 doesn't mean it works on macOS/arm64. I think it would be simpler and safer to just return "file n contains invalid label at offset x" as `string` from `post-rs` and then wrap it in the returned error:

```suggestion
	case C.VerifyResult_Invalid:
		return fmt.Errorf("%w: %s", ErrInvalidPos, C.GoString(result.error))
```

Reply:
All C code is dangerous :upside_down_face:. What you propose works for verifying POS, but wouldn't work for verifying proof - we need the index there.

Reply:
I think the issue I have with this approach is that the return value of the verify function is essentially a `void *` (or at least of the field `anon0` in the returned object), but I don't think it needs to be. Neither here nor in the case of verifying proof. The return value for `C.verify_pos` can just be (in go syntax):

```go
type pos_result struct {
	tag int   // the error code
	file string // only set when tag == C.VerifyResult_Invalid
	offset int // only set when tag == C.VerifyResult_Invalid
}
```

and the return value for `C.verify_proof` can instead be:

```go
type proof_result struct {
	tag int   // the error code
	index_id int // only set when tag == C.VerifyResult_InvalidIndex
}
```

Without a generic approach the object-byte-cast wouldn't be necessary 🙂.

Reply:
Maybe we can just give the `anon0` field a name? Then I'd already be happy 🙂

Reply:
The header is autogenerated with https://github.com/mozilla/cbindgen. I looked but I didn't find a way to force it to name the union :(